### ISSUE TITLE
PlayerTeleportEvent error

### ISSUE THREAD

**Issue by SlimeDog** [konsolas/AAC-Issues#1668]
### Issue description

Several times after server start-up, no players connected.
```
[16:16:19] [Server thread/ERROR]: Could not pass event PlayerTeleportEvent to AAC v4.0.7
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:316) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:66) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:507) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:492) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.craftbukkit.v1_13_R2.entity.CraftPlayer.teleport(CraftPlayer.java:602) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.craftbukkit.v1_13_R2.entity.CraftEntity.teleport(CraftEntity.java:311) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:256) [Citizens-2.0.25-b1738.jar:?]
        at net.citizensnpcs.npc.CitizensNPC.load(CitizensNPC.java:150) [Citizens-2.0.25-b1738.jar:?]
        at net.citizensnpcs.api.npc.SimpleNPCDataStore.loadInto(SimpleNPCDataStore.java:59) [Citizens-2.0.25-b1738.jar:?]
        at net.citizensnpcs.Citizens$3.run(Citizens.java:331) [Citizens-2.0.25-b1738.jar:?]
        at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftTask.run(CraftTask.java:81) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:392) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:889) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:831) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:729) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IllegalArgumentException: Unsupported protocol version: 404
        at me.konsolas.aac.aQ.a(aQ.java:46) ~[?:?]
        at me.konsolas.aac.aQ.a(aQ.java:18) ~[?:?]
        at me.konsolas.aac.V.<init>(V.java:1) ~[?:?]
        at me.konsolas.aac.bO.a(bO.java:12) ~[?:?]
        at me.konsolas.aac.bu.b(bu.java:88) ~[?:?]
        at me.konsolas.aac.bu.a(bu.java:218) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:312) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        ... 16 more
```

### Core information
**Server version**: Spigot 1.13.2 final

**AAC version**: 4.0.7

**ProtocolLib version**: 4.5.0-b9

**Plugin list**: 

### Special environment information
**ViaVersion version**: 2.1.3-b271

**ProtocolSupport version**: n/a

**ViaBackwards version**: n/a

**BungeeCord plugins**: n/a

### Configuration

Default except:
```
fightspeed:
  enabled: false
...
hitbox:
  max_reach: 14.1
```
in particular:
```
move:
  enabled: true

  # Should AAC check players asynchronously? This option reduces load on the main server thread by around 25%
  async: false

  # Move works by simulating many different input possibilities and checking if any match the actual movement.
  # Different Minecraft versions have different movement, so Move needs each player's client version to work correctly.
  # If you use ProtocolSupport, ViaVersion, or if you don't use any protocol hack plugins, "auto" will select
  # the appropriate client version for each player using the server version, or ProtocolSupport/ViaVersion's APIs.
  # Otherwise, you can force Move to check multiple different client versions, though this will hurt performance.
  simulate_versions:
    - "auto"
```
404 is the protocol version for 1.13.2.

Same configuration works fine on Spigot 1.14.4 (Jenkins 2328) latest.

**Comment by SlimeDog** [comment]
Same error several times after a player connected with 1.14.4 client to Spigot 1.13.2 server.

**Comment by Janmm14** [comment]
Which protocol version does it show for the 1.14.4 player?

**Comment by SlimeDog** [comment]
Where do I find/verify that information?

**Comment by Janmm14** [comment]
you say the same error occurs for 1.14.4 players on 1.13.2 server. So does it show the same number in the error?

**Comment by SlimeDog** [comment]
Same error after player connection:
```
[17:03:50] [Server thread/ERROR]: Could not pass event PlayerTeleportEvent to AAC v4.0.7
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:316) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:66) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:507) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:492) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.craftbukkit.v1_13_R2.entity.CraftPlayer.teleport(CraftPlayer.java:602) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.craftbukkit.v1_13_R2.entity.CraftEntity.teleport(CraftEntity.java:311) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:256) [Citizens-2.0.25-b1738.jar:?]
        at net.citizensnpcs.EventListen.spawn(EventListen.java:647) [Citizens-2.0.25-b1738.jar:?]
        at net.citizensnpcs.EventListen.respawnAllFromCoord(EventListen.java:617) [Citizens-2.0.25-b1738.jar:?]
        at net.citizensnpcs.EventListen.access$000(EventListen.java:100) [Citizens-2.0.25-b1738.jar:?]
        at net.citizensnpcs.EventListen$1.run(EventListen.java:148) [Citizens-2.0.25-b1738.jar:?]
        at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftTask.run(CraftTask.java:81) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:392) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:889) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:831) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:729) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IllegalArgumentException: Unsupported protocol version: 404
        at me.konsolas.aac.aQ.a(aQ.java:46) ~[?:?]
        at me.konsolas.aac.aQ.a(aQ.java:18) ~[?:?]
        at me.konsolas.aac.V.<init>(V.java:1) ~[?:?]
        at me.konsolas.aac.bO.a(bO.java:12) ~[?:?]
        at me.konsolas.aac.bu.b(bu.java:88) ~[?:?]
        at me.konsolas.aac.bu.a(bu.java:218) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:312) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        ... 17 more
```

**Comment by Janmm14** [comment]
Ok. In that case I suggest reverting the config option back to 1.13

**Comment by SlimeDog** [comment]
OK. I'll probably just leave it as-is, since the client can connect, and I definitely want to use the new "auto" config. I imagine there will be an update soon. :)

**Comment by Janmm14** [comment]
I would not do that, because this error likely prevents the move check form running.

**Comment by SlimeDog** [comment]
Ah, OK. Thanks. I'll make the change.

**Comment by Janmm14** [comment]
write "1.13" in config, jsut like it was before

**Comment by SlimeDog** [comment]
Done. Thanks.

**Comment by SlimeDog** [comment]
Same errors, with this configuration
```
move:
  enabled: true

  # Should AAC check players asynchronously? This option reduces load on the main server thread by around 25%
  async: false

  # Move works by simulating many different input possibilities and checking if any match the actual movement.
  # Different Minecraft versions have different movement, so Move needs each player's client version to work correctly.
  # If you use ProtocolSupport, ViaVersion, or if you don't use any protocol hack plugins, "auto" will select
  # the appropriate client version for each player using the server version, or ProtocolSupport/ViaVersion's APIs.
  # Otherwise, you can force Move to check multiple different client versions, though this will hurt performance.
  simulate_versions:
#   - "auto"
# e.g. Force Move to check only for 1.8 and 1.13 movement
#   - "1.8"
    - "1.13"
```

**Comment by konsolas** [comment]
Apologies, looks like I missed out the two most important protocol versions for 1.13

This is fixed in 4.0.7-b1

**Comment by SlimeDog** [comment]
Confirmed fixed in AAC 4.0.7-b1

Thanks.
